### ISSUE TITLE
missleading err msg "Multiple users with that username returned. Please use 'cf curl' with specific origin instead"

### ISSUE THREAD

**Issue by syslxg** [cloudfoundry/cli#1097]
## Command  

cf delete-user foo@bar

## What occurred
I get output:
```
Error deleting user foo@bar
Multiple users with that username returned. Please use 'cf curl' with specific origin instead
```
## What you expected to occur

I should get something more clear.
and the word `origin` in the err msg above really confused me.  It suggests something like ` cf curl -X DELETE /v2/users/id?origin=ldap` to me. but cc api does not support that `origin` param.

## CLI Version  

6.25.0

## CC API Endpoint Version  

(API version: 2.58.0)

**Comment by dkoper** [comment]
Hi @syslxg 

I understand the issue.
Have you figured out how to delete the user, and based on that, have a better suggestion for this message?
How about:
```
Error deleting user foo@bar
Multiple users with that username returned. Please use 'cf curl' to delete the user by guid instead.
```

I'd prefer not to include the exact UAA APIs here as with a new API version we would likely forget to update the message.

**Comment by syslxg** [comment]
@dkoper , that is fine with me.

**Comment by dkoper** [comment]
@syslxg Can you confirm whether this issue is resolved with our edge binary?
https://github.com/cloudfoundry/cli#edge-binaries

**Comment by wgu** [comment]
This turns out to be a wrong message to my test. `cf delete-user` != `cf curl -X DELETE`. The former will remove the specified user from both ccdb and uaa while the latter is merely an API call to cc. After a `cf delete-user` the login with that deleted account will report user not exist. But after a `cf curl -X DELETE` the account will still be able to login.

**Comment by dkoper** [comment]
@wgu Ah yes, I see what you mean.
Any suggestion on what would be a useful message?
Also, why did you get into this situation in the first place? I remember discussing such a case (same username in different origins) with the UAA PM when we added the `--origin` flag and could not come up with a use case for people to create same-named users.

**Comment by wgu** [comment]
This actually was a wrong operation which led to the issue. And yes, I think the ultimate issue is we are allowing users from different "origin"s to have same name. Would that be necessary? To me the platform should allow only 1 idp (origin). But in the case of an external idp co-existing with the internal one, would it be acceptable to keep the username unique as well? If one sees a conflict when trying to create a same username, the option could be either change the name or deal with the existing one first. Agreed that I don't see a use case for a same name account to server different actual users.

**Comment by dkoper** [comment]
Today's release (6.26.0) has the message that turns out not sufficient.

Whether users from different origins should be allowed to have the same name, or 1 external idp co-existing with UAA's (and if/how that is enforcible) is a question for the UAA PM to comment on. -> @sreetummidi?

**Comment by sreetummidi** [comment]
Usernames can be duplicated in two origins. The combination of username and origin and username should be unique and is enforced by UAA

**Comment by dkoper** [comment]
@jbpivotal & I iterated a few times over improving this error message, but couldn't find a simple way of explaining what to do. I felt any reference to `uaac` or `cf curl` would require detailed API documentation on how to use them.

What it boils down to though, is that the cf CLI (for now at least) does not have support for user deletion and role management for users that require their origin to be specified to disambiguate from other users with the same username.
We'll change the error message to simply say "The user exists in multiple origins.".

**Comment by dkoper** [comment]
Addressed in cf CLI 6.33.0.
